### ISSUE TITLE
Make quiver work offline

### ISSUE THREAD

**Issue by varkor** [varkor/quiver#157]
I often find myself using **quiver** without an internet connection, but this is less convenient for people who have not downloaded a local copy of the tool themselves. It would be useful to make **quiver** work offline by default.

**Comment by paolobrasolin** [comment]
Hi @varkor! I sketched a simple, cheap solution to this in #176.

No changes to existing JS code are required.

Testing it requires nothing more than locally serving the `/src` folder.

There are two moving parts:
* a **webapp manifest**, which is recognized by the browser and allows for "installation"; this simply means creating links (on both desktop and mobile) that open the webapp in a dedicated branded browser window with minimal or no interface except the web view
* a **service worker** that caches all requests and effectively allows the webapp to work offline, whether it's installed or not

The combined effect gives the feeling of having a native app (which works offline but also updates every time you go online), providing something that effectively behaves like a mobile app, and solving #44 too for desktops.

If you like this I can clean it up and prepare it for merge.

**Comment by varkor** [comment]
Hi @paolobrasolin, thanks for this! This does look like the simplest approach to implementing this feature. I think I'd be happy with this solution; there's a question of whether it'd be better to use Google's CDN, or host the workbox-sw files along with the other source files, but I think there's probably little drawback to using the CDN.

**Comment by paolobrasolin** [comment]
Since you're "vendoring" `KaTeX`, we might as well do that for `workbox` too.

`workbox-sw` secretly lazy-fetches dependencies from the CDN at runtime, but it's not difficult to "vendor" all `workbox` dependencies. I added a command to the `Makefile` to do just that. I also added a command to produce the mandatory icons for the webapp, so you don't have to keep those in-repo.

A tiny QOL improvement would be to detect whether we're deploying to prod or dev in order to enable `workbox` debug and maybe append a `[DEV]` tag to the name of the installed app. However that's a lot of extra complexity for very little benefit. Maybe in the future!

I think #176 is ready! 😄
